sdk/core/Azure.Core.TestFramework/src/Instrumentation/ManagementInterceptor.cs

```diff
@@ -21,8 +21,35 @@ public ManagementInterceptor(ClientTestBase testBase)
 
         public void Intercept(IInvocation invocation)
         {
+            bool modifiedAskToWait = false;
+
+            if (IsLro(invocation.Method.ReturnType))
```

Comment:
Why doesn't the LRO interceptor inside InstrumentResultInterceptor work for this scenario?

Reply:
Because the WaitForCompletion is called inside an instrumented method, so the LRO object is constructed and immediately that method is called and no chance to instrument the LRO object.